app/src/main/java/com/github/se/polyfit/data/api/SpoonacularAPI.kt

```diff
@@ -126,7 +125,7 @@ class SpoonacularApiCaller {
     val meal = MutableLiveData<Meal>()
     meal.postValue(Meal.default())
 
-    scope.launch {
+    GlobalScope.launch {
```

Comment:
This could lead to some issues later on as calling GlobalScope.launch makes it so that the coroutine will never shut down. Although it works from now since this coroutine doen't have any sense of the lifecyle of the activity or feature it is implemented in it will stay running in the background and could cause some issues. 


some alternative could be linking it to a viewModel and call the viewmodelScope.launch coroutine or even lifecycleScope.launch
Another idea would be to create you're own scope this would allow more control if needed.

not needed for now but put a TODO on it to modify it later

Reply:
The coroutine will shut down because the functions launched within it are guaranteed to end at some point. For example, if the API does not respond within a certain amount of time, it will trigger a timeout, causing every sub-function to return. The issue is that Spoonacular is not run within a single ViewModel. For instance, if it is launched in one ViewModel and that ViewModel is stopped, then its coroutine will also stop. That is why I resorted to a global scope to avoid having any issues if anything happened with the viewmodel.

But creating my own scope could be an idea. I will need to look more into it.